config/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+//! foundry configuration.
+use ethers_core::types::Address;
+use ethers_solc::{remappings::Remapping, ProjectPathsConfig};
+use figment::{
+    providers::{Env, Format, Serialized, Toml},
+    value::{Dict, Map},
+    Figment, Metadata, Profile, Provider,
+};
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, path::PathBuf};
+
+/// Foundry configuration
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub struct Config {
+    /// The selected profile. **(default: _default_ `default`)**
+    ///
+    /// **Note:** This field is never serialized nor deserialized. When a
+    /// `Config` is merged into a `Figment` as a `Provider`, this profile is
+    /// selected on the `Figment`. When a `Config` is extracted, this field is
+    /// set to the extracting Figment's selected `Profile`.
+    #[serde(skip)]
+    pub profile: Profile,
+    /// path of the source contracts dir, like `src` or `contracts`
+    pub src: PathBuf,
+    /// path of the test dir
+    pub test: PathBuf,
+    /// path to where artifacts shut be written to
+    pub out: PathBuf,
+    /// all library folders to include, `lib`, `node_modules`
+    pub libs: Vec<PathBuf>,
+    /// whether to enable cache
+    pub cache: bool,
+    /// concrete solc version to use if any,
+    pub solc_version: Option<Version>,
```

Comment:
So this would trigger --no-autodetect?

Reply:
yeah but it's setting solc version and `autodetect` some what overlap,
so I think solc_version should take precedence over `auto-detect`.
by default autodetect is true.